src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs

```diff
@@ -478,6 +461,15 @@ private static Document GetReportedDocument(Diagnostic diagnostic, ImmutableDict
             return currentSolution;
         }
 
+        /// <summary>
+        /// Try to merge the changes between <paramref name="newDocument"/> and <paramref name="oldDocument"/> into <paramref name="cumulativeChanges"/>.
+        /// If there is any conflicting change in <paramref name="newDocument"/> with existing <paramref name="cumulativeChanges"/>, then the original <paramref name="cumulativeChanges"/> are returned.
+        /// Otherwise, the newly merged changes are returned.
+        /// </summary>
+        /// <param name="oldDocument">Base document on which FixAll was invoked.</param>
+        /// <param name="newDocument">New document with a code fix that is being merged.</param>
+        /// <param name="cumulativeChanges">Existing merged changes from other batch fixes into which newDocument changes are being merged.</param>
```

Comment:
you can use intervaltree for cumulative changes.

http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Collections/IntervalTree%25601.cs,191f1141194b2e6a